### DISCUSSION TITLE
Supported databases

### DISCUSSION THREAD

**Discussion by infused-kim** [directus/directus#2771] [Archive]
I just discovered this software and it looks absolutely amazing.

But I can't figure out what are the supported databases? It seems like the software itself requires mysql to run.

But is it able to connect to other SQL types? Particularly I am interested in MS SQL Server.

I read in the documentation that it is using the Zend DB framework. So theoretically that should be able to connect to it, right?

I also saw that there was a vote on which DB to support next, including mongo, postgres and ms sql. Mongo won, but once again I am not clear if this is the DB that's required for the internal storage of data of directus or for the final DB where the data is stored?

**Comment by rijkvanzanten** [comment]
Heya! Right now, only MySQL is supported. While Zend was an OK abstraction around SQL itself, it didn't offer everything we need to run Directus across multiple database drivers (things like schema reading/manipulation). We're currently moving the stack over to Laravel currently to be able to support multiple databases.

**Reply by benhaynes** [reply]
To add on to this, with the Laravel and Node ports in v9, we're looking to add support for PostgreSQL, MS-SQL, SQLite, and more...

**Reply by OfficialOzzy** [reply]
Hi @benhaynes ,

When is version 9 supposed to be released?

**Reply by benhaynes** [reply]
RC this week, Prod in the next 2-4 weeks.

**Comment by shrike71** [comment]
Hi All,

Also think this is amazing, but we're a PostGres house throughout... it would be nice to know when Postgres compatibility is going to be available?

Thanks & keep up the great work

shrike71

**Reply by benhaynes** [reply]
In a few weeks with the Directus 9 release.

**Reply by shrike71** [reply]
Hi Ben,

Thanks for the info!

shrike71

**Reply by shrike71** [reply]
Hi Ben,

We've got a pressing issue (read immediate!) with developing a documentation portal for one of our apps. If we had to use MySQL as an "appliance" install with Directus 8, might you provide a guideline or a set of tools if we had to migrate to PostGres when v,9 comes out? 

We're mainly a PostGres house, and although we have to run one or two MS SQL Servers for COTS applications, we would prefer not to have to introduce MySQL into the support mix also, or at least introduce it only for a short time.

Thanks,

Shrike71

**Reply by benhaynes** [reply]
You can use v9... but there is no official migration for the system settings/tables.

https://github.com/directus/next

**Reply by shrike71** [reply]
Hi Ben,

Just noticed that you've uploaded a new container covering v9 to Docker Hub. Congratulations on the milestone! Might you be able to post a sample docker-compose.yml, or changes to the existing example that might demonstrate a connection to a local or AWS RDS PostGres instance?

Thank you

Shrike71

**Reply by benhaynes** [reply]
Thanks! As for the sample file, I defer to @WoLfulus on where we would add this...
